graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/Federation.java

```diff
@@ -43,12 +41,14 @@ public static SchemaTransformer transform(
   public static SchemaTransformer transform(
       final TypeDefinitionRegistry typeRegistry, final RuntimeWiring runtimeWiring) {
     final boolean queryTypeShouldBeEmpty = ensureQueryTypeExists(typeRegistry);
+
+    boolean isFederation2 = isFederation2(typeRegistry);
     RuntimeWiring newRuntimeWiring =
-        ensureFederationDirectiveDefinitionsExist(typeRegistry, runtimeWiring);
+        ensureFederationDirectiveDefinitionsExist(typeRegistry, runtimeWiring, isFederation2);
     final GraphQLSchema original =
         new SchemaGenerator()
             .makeExecutableSchema(generatorOptions, typeRegistry, newRuntimeWiring);
-    return transform(original, queryTypeShouldBeEmpty);
+    return transform(original, queryTypeShouldBeEmpty).setFederation2(isFederation2);
```

Comment:
Because we get `isFederation` from the `typeRegistry`, this does not apply to `Federation.transform(GraphQLSchema)`. I'm not sure who's using that though. DGS uses [`Federation.transform(typeRegistry)`](https://github.com/netflix/dgs-framework/blob/595f2a8762a53f321c9285b331bf9196958e2cc8/graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsSchemaProvider.kt#L134) and graphql-kotlin [uses its own implementation](https://github.com/ExpediaGroup/graphql-kotlin/blob/de02352dc8af5bc6bb78b90d8d8f4908988999f6/generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/FederatedSchemaGeneratorHooks.kt#L82).